### PR TITLE
Remove sqlsmith extension

### PR DESCRIPTION
**Pull request by Tmonster**
Removing the SQL Smith extension. Have a repo up here so the extension can go out of tree https://github.com/Tmonster/duckdb-sqlsmith . Before this gets merged, some other tasks need to happen, This PR is being opened to check CI

- [x] Ownership of the duckdb-sqlsmith repo should move to duckdb
- [ ]  ~~More fuzzer scripts should move to the sqlsmith repo.~~ Will do this in another PR
- [x]  CI workflows in https://github.com/duckdblabs/duckdb-fuzzer-ci should be checked to make sure they work with sql-smith out of tree.  -- For now they will since no scripts are being moved out of tree. Has been done with https://github.com/duckdblabs/duckdb-fuzzer-ci/pull/24

### CODE REVIEW

File: src/include/duckdb/main/extension_entries.hpp [hunk 1]
```diff
@@ -44,8 +44,6 @@ static constexpr ExtensionFunctionEntry EXTENSION_FUNCTIONS[] = {
     {"from_json_strict", "json", CatalogType::SCALAR_FUNCTION_ENTRY},
     {"from_substrait", "substrait", CatalogType::TABLE_FUNCTION_ENTRY},
     {"from_substrait_json", "substrait", CatalogType::TABLE_FUNCTION_ENTRY},
-    {"fuzz_all_functions", "sqlsmith", CatalogType::TABLE_FUNCTION_ENTRY},
-    {"fuzzyduck", "sqlsmith", CatalogType::TABLE_FUNCTION_ENTRY},
```
Comment (Mytherin) [inline src/include/duckdb/main/extension_entries.hpp:48]: I think these shouldn't be removed

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Great idea - LGTM. One minor comment

File: .github/config/out_of_tree_extensions.cmake [hunk 2]
```diff
@@ -100,6 +100,11 @@ duckdb_extension_load(sqlite_scanner
         GIT_TAG 50b7870be099186f195bc72bac5e9e11247ee2f9
         )
 
+duckdb_extension_load(sqlsmith)
```
Comment (carlopi) [inline .github/config/out_of_tree_extensions.cmake:103]:
```suggestion
duckdb_extension_load(sqlsmith
```
I think like this it should work

**Review by carlopi — CHANGES_REQUESTED** [review]
Minor typo, plus I think duckdblabs/duckdb-fuzzer-ci should be moved from BUILD_SQLSMITH=1 to point to the now out of tree extension.

Unsure if there are other places (that we care about) that are using the BUILD_SQLSMITH=1 syntax.

**Review by carlopi — APPROVED** [review]
Cool, thanks.

Should this go into feature OR directly into main? I would not be opposed for this to go in main. Either way I think fuzzer should directly use duckdb/duckdb_sqlsmith, and same development should be made in the separate repo.

I realized something that had not seen before, we could change the Makefile so that BUILD_SQLSMITH=1 becomes equivalent to append, at the Makefile level, sqlsmith.cmake to the list of cmake files that will get checked.

That would be the way to support migration for BUILD_X.
In the case of SQLSMITH is not super relevant, but other extensions are harder to separate.

**Comment by Tmonster** [comment]
I'm also fine with this going into main, but given that main is currently being used to track v1.0.0 I don't think this should go in until all bug fix releases are out or v1.0.0 has it's own branch for future bug fixes

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**